pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -54,37 +54,35 @@ def __init__(self, min_steps: int, max_steps: int):
         self._dataloader_idx: Optional[int] = None
         self._warning_cache: WarningCache = WarningCache()
         self._epoch_output: Optional[List[List[STEP_OUTPUT]]] = None
+        self._iterable_dataset_samplers_state_dict = {}
 
     @property
     def batch_idx(self) -> int:
         """Returns the current batch index (within this epoch)"""
         return self.iteration_count
 
+    @property
+    def total_epoch_completed(self) -> int:
+        return self.progress.total.completed
+
+    @property
+    def total_optimizer_step(self) -> int:
+        return self.batch_loop.total_optimizer_step
+
+    @property
+    def current_batch_seen(self) -> int:
+        return self.progress.batch.current.completed
+
     @property
     def done(self) -> bool:
         """Returns whether the training should be stopped.
         The criteria are that the number of steps reached the max steps,
         the last batch is reached or the trainer signals to stop (e.g. by early stopping).
         """
-        max_steps_reached = self.max_steps is not None and self.global_step >= self.max_steps
+        max_steps_reached = self.max_steps is not None and (self.total_optimizer_step) >= self.max_steps
```

Comment:
max steps in the trainer does not represent optimizer steps (today). It represents how many times training_step() was called (roughly). If we want to change that I think it should be done in a separate PR with a proper evaluation of what the effects are.